server/aesdsocket.c

```diff
@@ -671,14 +719,20 @@ void socket_termination_signal_handler(int signo)
     perror("Failed on shutdown");
     syslog(LOG_ERR,"Could not close socket fd in signal handler: %s",strerror(errno));
   }
-  
+ 
+#if USE_AESD_CHAR_DEVICE
```

Comment:
@Rishab-Shah think about how you can minimize the number of build switches you need for your timer.  You sould be able to leave your code compiled in and just not start the timer with the `#if` logic, hopefully in one spot.